data/shared/src/main/scala/sigma/ast/methods.scala

```diff
@@ -303,6 +303,18 @@ case object SIntMethods extends SNumericTypeMethods {
 case object SLongMethods extends SNumericTypeMethods {
   /** Type for which this container defines methods. */
   override def ownerType: SMonoType = SLong
+
+  lazy val DecodeNBitsMethod: SMethod = SMethod(
+    this, "DecodeNBits", SFunc(this.ownerType, SBigInt), 8, FixedCost(JitCost(5)))
+    .withInfo(PropertyCall, "Consider this Long value as nbits-encoded BigInt value and decode it to BigInt")
+
+  protected override def getMethods(): Seq[SMethod] = {
+    if (VersionContext.current.isV6SoftForkActivated) {
+      super.getMethods() ++ Seq(DecodeNBitsMethod)
+    } else {
+      super.getMethods()
+    }
+  }
```

Comment:
Unfortunately, this simple versioning is very fragile for global data structures (which is `object SLongMethods`). 
It may be used from different contexts and even if a correct `VersionContext.current` is provided, the results of this method is saved in other global data structures (i.e. see `private lazy val _methodsMap`).
I think it is better to implement explicit versioning of this metadata.